tests/components/evohome/test_storage.py

```diff
@@ -0,0 +1,133 @@
+"""The tests for evohome storage load & save."""
+
+from datetime import datetime
+from typing import Any, Final
+from unittest.mock import patch
+
+import pytest
+
+from homeassistant.components.evohome import (
+    CONF_PASSWORD,
+    CONF_USERNAME,
+    DOMAIN,
+    EvoBroker,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.setup import async_setup_component
+import homeassistant.util.dt as dt_util
+
+from .conftest import mock_get
+from .const import (
+    ACCESS_TOKEN,
+    ACCESS_TOKEN_EXPIRES_DTM,
+    ACCESS_TOKEN_EXPIRES_STR,
+    DIFF_EMAIL_ADDRESS,
+    REFRESH_TOKEN,
+    SAME_EMAIL_ADDRESS,
+    SESSION_ID,
+)
+
+TEST_CONFIG: Final = {
+    CONF_USERNAME: SAME_EMAIL_ADDRESS,
+    CONF_PASSWORD: "password",
+}
+
+
+TEST_DATA: Final = {
+    "00": None,
+    "01": {},
+    "10": {
+        "username": SAME_EMAIL_ADDRESS,
+        "refresh_token": REFRESH_TOKEN,
+        "access_token": ACCESS_TOKEN,
+        "access_token_expires": ACCESS_TOKEN_EXPIRES_STR,
+    },
+    "11": {
+        "username": SAME_EMAIL_ADDRESS,
+        "refresh_token": REFRESH_TOKEN,
+        "access_token": ACCESS_TOKEN,
+        "access_token_expires": ACCESS_TOKEN_EXPIRES_STR,
+        "user_data": {"sessionId": SESSION_ID},
+    },
+}
+
+
+@patch("evohomeasync2.broker.Broker.get", mock_get)
+@pytest.mark.parametrize("idx", TEST_DATA)
+async def test_load_auth_tokens_same(
+    hass: HomeAssistant, hass_storage: dict[str, Any], idx: str
+) -> None:
+    """Test restoring authentication tokens when matching account."""
+
+    hass_storage[DOMAIN] = {
+        "version": 1,
+        "minor_version": 1,
+        "key": "evohome",
+        "data": TEST_DATA[idx],
+    }
+
+    assert await async_setup_component(hass, DOMAIN, {DOMAIN: TEST_CONFIG})
+    await hass.async_block_till_done()
+
+    evo_broker: EvoBroker = hass.data[DOMAIN]["broker"]
```

Comment:
Don't interact with integration details like things in hass.data. Patch the library client and assert things about the client Mock.